controllers/capabilities/mapping_rules.go

```diff
@@ -105,9 +114,12 @@ func (t *ProductThreescaleReconciler) syncMappingRules(_ interface{}) error {
 
 func (t *ProductThreescaleReconciler) processNotDesiredMappingRules(notDesiredList []threescaleapi.MappingRuleItem) error {
 	for _, mappingRule := range notDesiredList {
-		err := t.productEntity.DeleteMappingRule(mappingRule.ID)
-		if err != nil {
-			return err
+		if mappingRule.HTTPMethod == "" || mappingRule.Pattern == "" || mappingRule.Pattern == "/" {
+			t.logger.V(1).Info("processNotDesiredMappingRules, delete", "HTTPMethod", mappingRule.HTTPMethod, "Pattern", mappingRule.Pattern, "MetricID", strconv.FormatInt(mappingRule.MetricID, 10))
```

Comment:
As suggested on our chat, I would not go into metricID here at all. Instead, pull the mappingRule.Position.